packages/react-select/src/accessibility/index.js

```diff
@@ -1,92 +1,136 @@
   tabSelectsValue?: boolean,
 };
-export type ValueEventContext = { value: string, isDisabled?: boolean };
-
-export const instructionsAriaMessage = (
-  event: string,
-  context?: InstructionsContext = {}
-) => {
-  const { isSearchable, isMulti, label, isDisabled, tabSelectsValue } = context;
-  switch (event) {
-    case 'menu':
-      return `Use Up and Down to choose options${
-        isDisabled ? '' : ', press Enter to select the currently focused option'
-      }, press Escape to exit the menu${
-        tabSelectsValue
-          ? ', press Tab to select the option and exit the menu'
-          : ''
-      }.`;
-    case 'input':
-      return `${label ? label : 'Select'} is focused ${
-        isSearchable ? ',type to refine list' : ''
-      }, press Down to open the menu, ${
-        isMulti ? ' press left to focus selected values' : ''
-      }`;
-    case 'value':
-      return 'Use left and right to toggle between focused values, press Backspace to remove the currently focused value';
-  }
+
+export type AriaSelectionType = SelectionContext & {
+  // first parameter passed in onChange
+  value?: ValueType,
 };
 
-export const valueEventAriaMessage = (
-  event: string,
-  context: ValueEventContext
-) => {
-  const { value, isDisabled } = context;
-  if (!value) return;
-  switch (event) {
-    case 'deselect-option':
-    case 'pop-value':
-    case 'remove-value':
-      return `option ${value}, deselected.`;
-    case 'select-option':
-      return isDisabled
-        ? `option ${value} is disabled. Select another option.`
-        : `option ${value}, selected.`;
-  }
+export type AriaLiveMessagesProps = {
+  onChange?: (value: ValueType, context: SelectionContext) => string,
+  onFocus?: (focused: OptionType, context: FocusedContext) => string,
+  onFilter?: (args: {
+    inputValue: string,
+    screenReaderMessage: string,
+  }) => string,
+  guidance?: (type: GuidanceType, context?: GuidanceContext) => string,
 };
 
-export const valueFocusAriaMessage = ({
-  focusedValue,
-  getOptionLabel,
-  selectValue,
-}: {
-  focusedValue: OptionType,
-  getOptionLabel: (option: OptionType) => string,
-  selectValue: OptionsType,
-}) =>
-  `value ${getOptionLabel(focusedValue)} focused, ${selectValue.indexOf(
-    focusedValue
-  ) + 1} of ${selectValue.length}.`;
-
-export const optionFocusAriaMessage = ({
-  focusedOption,
-  getOptionLabel,
-  options,
-}: {
-  focusedOption: OptionType,
-  getOptionLabel: (option: OptionType) => string,
-  options: OptionsType,
-}) =>
-  `option ${getOptionLabel(focusedOption)} focused${
-    focusedOption.isDisabled ? ' disabled' : ''
-  }, ${options.indexOf(focusedOption) + 1} of ${options.length}.`;
-
-export const resultsAriaMessage = ({
-  inputValue,
-  screenReaderMessage,
-}: {
-  inputValue: string,
-  screenReaderMessage: string,
-}) =>
-  `${screenReaderMessage}${
-    inputValue ? ' for search term ' + inputValue : ''
-  }.`;
+export const defaultAriaLiveMessages = {
```

Comment:
Make this type-safe with:

```ts
export const defaultAriaLiveMessages: AriaLiveMessagesProps = {
```

Reply:
Actually this wouldn't work as I expected.  Sorry, I'm from the TypeScript world, and I can't find a Flow-equivalent to TypeScript's `Required<T>` utility type.

Reply:
I may have spoken too soon (Forgive my wordiness); just found a use with `$Shape`:

What you could do is make `AriaLiveMessagesProps` all non-optional, and then `export $Shape<T>`:

```typescript
type RequiredAriaLiveMessagesProps = {
  onChange: (value: ValueType, context: SelectionContext) => string,
  onFocus: (focused: OptionType, context: FocusedContext) => string,
  onFilter: (args: {
    inputValue: string,
    resultsMessage: string,
  }) => string,
  guidance: (type: GuidanceType, context?: GuidanceContext) => string,
};

export type AriaLiveMessagesProps = $Shape<RequiredAriaLiveMessagesProps>

export const defaultAriaLiveMessages: $Exact<RequiredAriaLiveMessagesProps> = {
```

Reply:
I appreciate you taking a look at this PR.

It's worth mentioning that the @Methuselah96 has put a lot of time into converting the project from Flow to TypeScript and the plan is to make the move shortly after this PR is released.

Reply:
That sounds great.  Let me know if you need some help, or some :eyes: @Methuselah96 

Also, yes, I am actively testing this PR. 😄